nebula-logger/plugins/big-object-archiving/plugin/classes/LogEntryArchivePlugin.cls

```diff
@@ -0,0 +1,225 @@
+        }
+        this.saveLogEntryArchives(BIG_OBJECT_IMMEDIATE_LOGGER_SAVE_METHOD);
+    }
+
+    /**
+     * @description No-op method, required by the interface `LoggerPlugin.Batchable`
+     * @param  configuration The instance of `LoggerPlugin__mdt` configured for this specific plugin
+     * @param  input The instance of `LoggerBatchableContext`, provided by the logging system
+     */
+    @SuppressWarnings('PMD.EmptyStatementBlock')
+    public void finish(LoggerPlugin__mdt configuration, LoggerBatchableContext input) {
+        // No-op
+    }
+
+    // Triggerable interface method
+    /**
+     * @description Handles converting Logger's buffer of `LogEntryEvent__e` records into `LogEntryArchive__b` records
+     *              for any user with the included custom save method 'BIG_OBJECT'
+     * @param  configuration The instance of `LoggerPlugin__mdt` configured for this specific plugin
+     * @param  input The instance of `LoggerTriggerableContext`, provided by the logging system
+     */
+    public void execute(LoggerPlugin__mdt configuration, LoggerTriggerableContext input) {
+        if (input.sobjectType != Schema.LogEntryEvent__e.SObjectType) {
+            return;
+        }
+        Set<String> bigObjectSaveMethods = new Set<String>{ BIG_OBJECT_IMMEDIATE_LOGGER_SAVE_METHOD, BIG_OBJECT_QUEUEABLE_LOGGER_SAVE_METHOD };
+
+        if (
+            input.triggerOperationType == TriggerOperation.BEFORE_INSERT &&
+            bigObjectSaveMethods.contains(Logger.getUserSettings().DefaultSaveMethod__c) == false
+        ) {
+            return;
+        }
+
+        List<LogEntryEvent__e> logEntryEvents = this.filterLogEntryEventsToSave(input);
+        for (LogEntryEvent__e logEntryEvent : logEntryEvents) {
+            LOG_ENTRY_ARCHIVES_TO_SAVE.add(new LogEntryArchiveBuilder(logEntryEvent).getLogEntryArchive());
+        }
+
+        if (LOG_ENTRY_ARCHIVES_TO_SAVE.isEmpty() == true) {
+            return;
+        }
+
+        String saveMethodName;
+        switch on input.triggerOperationType {
+            when BEFORE_INSERT {
+                saveMethodName = Logger.getUserSettings().DefaultSaveMethod__c;
+            }
+            when AFTER_INSERT {
+                saveMethodName = BIG_OBJECT_QUEUEABLE_LOGGER_SAVE_METHOD;
+            }
+        }
+        this.saveLogEntryArchives(saveMethodName);
+
+        Logger.flushBuffer();
+    }
+
+    private List<LogEntry__c> requeryLogEntriesToArchive(List<LogEntry__c> logEntries) {
+        String logLookupRelationshipName = Schema.LogEntry__c.Log__c.getDescribe().getName().removeEnd('__c') + '__r';
+        List<String> logFieldsToQuery = new List<String>();
+        for (String fieldName : Schema.Log__c.SObjectType.getDescribe().fields.getMap().keySet()) {
+            logFieldsToQuery.add(logLookupRelationshipName + '.' + fieldName);
+        }
+        String closedByUsernameField =
+            logLookupRelationshipName +
+            '.' +
```

Comment:
this looks like it could be cleaned up a bit, or at least have a TODO for drying it up

Reply:
Good call, there's a lot of duplicated code here - I'll try to move some of the duplicate logic here into a private method for reuse + cleanup.